utils/lib_sxng_data.sh

```diff
@@ -49,6 +51,15 @@ data.useragents() {
     dump_return $?
 }
 
+data.locales() {
+    (   set -e
+        pyenv.activate
+        build_msg DATA "update searx/data/locales.json"
+        python searxng_extra/update/update_locales.py
+    )
```

Comment:
Just for my understanding, what do these brackets do? Do they just create a new "scope", so that anything running inside these brackets (inner function) will not affect the outer function (data.locales in this case)?

I've already seen this multiple times, so I'm wondering.

Reply:
> what do these brackets do? 

it wraps commands in a subshell / here in this example the option `set -e` is only active in the subshell commands:

[`set -e``](https://www.gnu.org/software/bash/manual/html_node/The-Set-Builtin.html) .. exit (sub-)shell immediately if a command returns a non-zero status.

With the construct:

```bash
(  set -x
   command1
   command2 # ends a non-zero-status
   command3
)
```

You assert that command3 is not called when command2 ends with a non-zero-status

Reply:
Good to know, thank you!